dali/imgcodec/decoders/decoder_test_helper.h

```diff
@@ -139,6 +187,20 @@ class CpuDecoderTestBase : public ::testing::Test {
     return ReadReference(src.get());
   }
 
+  /**
+   * @brief Get device_id for the Backend
+   */
+  int GetDeviceId() {
+    if constexpr(std::is_same<Backend, CPUBackend>::value) {
```

Comment:
I guess it's no big deal because it's a test header, but in general we refrain from using C++17 in headers. This one can stay though, since it is intended to be used only in tests.